File: backend/javascript/backend-javascript-nest/src/seed.js
'use strict';

const { hashPassword } = require('./passwords');
const { DuplicateUsernameError } = require('./errors');

const SEED_ITEMS = [
  { name: 'Alpha', description: 'First seeded item from PostgreSQL' },
  { name: 'Beta', description: 'Second seeded item for demo API' },
  { name: 'Gamma', description: 'Third item — multistack bootstrap' },
];

const SEED_USERNAME = 'user1';
const SEED_PASSWORD = 'password1';

async function seedData(store) {
  if ((await store.countItems()) === 0) {
    await store.insertItems(SEED_ITEMS);
  }

  if ((await store.findUserByUsername(SEED_USERNAME)) === null) {
    try {
      await store.insertUser(SEED_USERNAME, hashPassword(SEED_PASSWORD));
    } catch (error) {
      // Another replica seeded the same user first — nothing left to do.
      if (!(error instanceof DuplicateUsernameError)) {
        throw error;
      }
    }
  }
}

module.exports = { seedData, SEED_ITEMS, SEED_USERNAME, SEED_PASSWORD };

File: backend/python/backend-python-django/api/seed.py
from __future__ import annotations

from api.auth import hash_password
from api.models import Item, User

SEED_ITEMS = (
    ("Alpha", "First seeded item from PostgreSQL"),
    ("Beta", "Second seeded item for demo API"),
    ("Gamma", "Third item — multistack bootstrap"),
)

SEED_USERNAME = "user1"
SEED_PASSWORD = "password1"


def seed_data() -> None:
    if Item.objects.count() == 0:
        Item.objects.bulk_create(
            [Item(name=name, description=description) for name, description in SEED_ITEMS]
        )
    if not User.objects.filter(username=SEED_USERNAME).exists():
        User.objects.create(
            username=SEED_USERNAME,
            password_hash=hash_password(SEED_PASSWORD),
        )
